examples/Python/PET/interactive/spdhg_reconstruction.py

```diff
@@ -0,0 +1,310 @@
+    shutil.copytree('brain', 'working_folder/brain')
+    os.chdir('working_folder/brain')
+elif EXAMPLE == 'nema':
+    os.chdir('mMR')
+    noisy_data = pet.AcquisitionData('sino_f1g1d0b0.hs')
+else:
+    raise ValueError('unexpected example')
+
+if EXAMPLE in ['brain', 'thorax']:
+    # We will first create some simulated data from ground-truth images
+
+    # Read in images
+    image = pet.ImageData('emission.hv')
+    image_array = image.as_array()*.05
+    image.fill(image_array)
+    mu_map = pet.ImageData('attenuation.hv')
+    mu_map_array = mu_map.as_array()
+    # bitmap display of images
+    sliceno = image_array.shape[0]/2
+    plt.figure()
+    imshow(image_array[sliceno, :, :, ], title='emission image')
+
+    # save max for future displays
+    cmax = image_array.max()*.6
+
+    # create acquisition model
+    am = pet.AcquisitionModelUsingRayTracingMatrix()
+    am.set_num_tangential_LORs(5)
+    templ = pet.AcquisitionData('template_sinogram.hs')
+    am.set_up(templ, image)
+    # simulate some data using forward projection
+    acquired_data = am.forward(image)
+    acquisition_array = acquired_data.as_array()
+
+    # Display bitmaps of a middle sinogram
+    plt.figure()
+    imshow(acquisition_array[0, :, :, ], title='Forward projection')
+
+    # close all plots
+    plt.close('all')
+
+    data = acquired_data
+    background = data.copy()
+    background.fill(5)
+
+    array = data.as_array() + background.as_array()
+    noisy_array = numpy.random.poisson(array.astype('float64'))
+    max_counts = noisy_array.max()
+    print(' Maximum counts in the data: {}'.format(noisy_array.max()))
+    noisy_data = data.clone()
+    noisy_data.fill(noisy_array)
+elif EXAMPLE == 'nema':
+    norm_file = 'norm.n.hdr'
+    attn_file = 'mu_map.hv'
+    sino_file = 'sino_f1g1d0b0.hs'
+    rand_file = 'sino_randoms_f1g1d0b0.hs'
+    noisy_data = pet.AcquisitionData(sino_file)
+    randoms = pet.AcquisitionData(rand_file)
+    asm_norm = pet.AcquisitionSensitivityModel(norm_file)
+
+    am = pet.AcquisitionModelUsingRayTracingMatrix()
+    am.set_num_tangential_LORs(10)
+    # add it to the acquisition model
+    am.set_acquisition_sensitivity(asm_norm)
+    # Attenuation image
+    attn_image = pet.ImageData(attn_file)
+    asm_attn = pet.AcquisitionSensitivityModel(attn_image, am)
+    # converting attenuation into attenuation factors (see previous exercise)
+    asm_attn.set_up(noisy_data)
+    attn_factors = pet.AcquisitionData(noisy_data)
+    attn_factors.fill(1.0)
+    print('applying attenuation (please wait, may take a while)...')
+    asm_attn.unnormalise(attn_factors)
+    print('done')
+    # use these in the final attenuation model
+    asm_attn = pet.AcquisitionSensitivityModel(attn_factors)
+    # chain attenuation and normalisation
+    asm = pet.AcquisitionSensitivityModel(asm_norm, asm_attn)
+    # update the acquisition model etc
+    am.set_acquisition_sensitivity(asm)
+
+    am.set_background_term(randoms)
```

Comment:
I think this is incorrect for SPDHG. you will add it to the KullbackLeibler later.

Or, you add it here, and then use the `get_linear_model` below (which would be nice and clear in my opinion).

Reply:
@KrisThielemans I'm not the most entitled to comment on this (@mehrhardt  help), however the KullbackLeibler fidelity [gets passed](https://github.com/CCPPETMR/SIRF/blob/add_to_sirf_classes/examples/Python/PET/interactive/spdhg_reconstruction.py#L236) the fully configured `AcquisitionMethod`.  

Then the `direct` method in the `AcquisitionModel`  [calls](https://github.com/CCPPETMR/SIRF/blob/add_to_sirf_classes/src/xSTIR/pSTIR/STIR.py#L1134) `self.get_linear_acquisition_model().forward()`
 
I had originally added this example to #238 so that I'd avoid this discussion on this particular PR, which deals with the additions of methods to SIRF classes in order to be able to use the CIL optimisation framework. The reason this example has appeared back here is the abstract @dkazanc is preparing. I'm happy to strip this example out of the PR so that we can merge the PR asap and then correct it.

Reply:
well, let's not do crazy merges before the results are ok...

in your link, `AcquisitionModel,direct` just calls `forward`. I know we originally said that `direct` would do this, but we changed our mind...

For the future, I don't think it makes sense to let the user add the background to KL, and "hope" that `spdgh` does the appropriate thing. But I wouldn't worry about it now.

I still don't know if the current code is doing the wrong thing of course...